slug_test.go

```diff
@@ -1227,28 +1245,49 @@ func assertArchiveFixture(t *testing.T, slug *bytes.Buffer, got *Meta) {
 	}
 }
 
-func verifyFile(t *testing.T, path string, mode os.FileMode, expect string) {
-	info, err := os.Lstat(path)
+func verifyTimestamps(t *testing.T, src, dst string) {
+	sourceInfo, err := os.Lstat(src)
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("source file %q not found", src)
+	}
+
+	dstInfo, err := os.Lstat(dst)
+	if err != nil {
+		t.Fatalf("dst file %q not found", dst)
+	}
+
+	// archive/tar purports to round timestamps to the nearest second so that behavior
+	// is duplicated here to test the restored timestamps.
+	sourceModTime := sourceInfo.ModTime().Round(time.Second)
```

Comment:
I think we want `Truncate` instead, I'd expect `Round` to be flakey?

```suggestion
	sourceModTime := sourceInfo.ModTime().Truncate(time.Second)
```

Ideally we wouldn't round/truncate here, because `lchtimes.go` would set the exact timestamp?

Reply:
Let's have all the time precision related discussion here!

By using Round, I was trying to [test this behavior](https://github.com/golang/go/blob/b86e8dd0f3a27bc9577690523c9feeb25593bec7/src/archive/tar/writer.go#L83-L92)

We _could_ modify our own code to truncate the second during Pack, but that would not be backwards-compatible with existing slugs.

Looking back at your other comment on Lchtimes, it seems like that, too, should use Round() instead of Unix(), to keep the same compatibility. I will give this another think tomorrow with a fresh brain.

Reply:
OK with a fresh pair of eyes it seems to me that, if we presume that the slug was created by go-slug (not necessarily a safe assumption, but safe for the purposes of this PR) then Unix() is a safe way to express the times in Lchtimes, because the timestamp would have been rounded by the archive/tar header.

I've added some comments and review fixes in two new commits. Let me know if I've clarified the situation or if you think we'd better revisit that presumption!

Reply:
I suppose even if I chose to use subsecond precision during Lchtimes, it wouldn't matter since most of the time there would be no subseconds anyway!

Reply:
Ah, so `archive/tar` complicates the situation. I'm inclined to suggest that we explicitly ignore the behaviour (except for how it complicates testing) and implement the changes as though we could always handle full accuracy or always truncate/round. That way `go-slug` is directly making the timestamp accuracy guarantee and doesn't depend on what `archive/tar` chooses to do.